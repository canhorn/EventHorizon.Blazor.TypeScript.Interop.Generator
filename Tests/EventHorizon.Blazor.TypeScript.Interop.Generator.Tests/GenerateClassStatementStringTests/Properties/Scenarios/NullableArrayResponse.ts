declare module Examples {
  export class ExampleClass {
    readonly datasetObservable: Observable<number[]>;
    inspectableCustomProperties: Nullable<IInspectable[]>;
  }
}
