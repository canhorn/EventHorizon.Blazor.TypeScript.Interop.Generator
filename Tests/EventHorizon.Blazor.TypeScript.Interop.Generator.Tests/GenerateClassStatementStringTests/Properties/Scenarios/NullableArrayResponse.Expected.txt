/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<ExampleClass>))]
    public class ExampleClass : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties
        private Observable<decimal[]> __datasetObservable;
        public Observable<decimal[]> datasetObservable
        {
            get
            {
            if(__datasetObservable == null)
            {
                __datasetObservable = EventHorizonBlazorInterop.GetClass<Observable<decimal[]>>(
                    this.___guid,
                    "datasetObservable",
                    (entity) =>
                    {
                        return new Observable<decimal[]>() { ___guid = entity.___guid };
                    }
                );
            }
            return __datasetObservable;
            }
        }

        
        public IInspectable[] inspectableCustomProperties
        {
            get
            {
            return EventHorizonBlazorInterop.GetArrayClass<IInspectable>(
                    this.___guid,
                    "inspectableCustomProperties",
                    (entity) =>
                    {
                        return new IInspectable() { ___guid = entity.___guid };
                    }
                );
            }
            set
            {

                EventHorizonBlazorInterop.Set(
                    this.___guid,
                    "inspectableCustomProperties",
                    value
                );
            }
        }
        #endregion
        
        #region Constructor
        public ExampleClass() : base() { }

        public ExampleClass(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods

        #endregion
    }
}